crates/fuel-core/src/service/genesis.rs

```diff
@@ -150,6 +149,23 @@ pub async fn execute_genesis_block(
     Ok(result)
 }
 
+pub async fn recover_missing_tables_from_genesis_state_config(
+    watcher: StateWatcher,
+    config: &Config,
+    db: &CombinedDatabase,
+) -> anyhow::Result<()> {
+    let genesis_block = create_genesis_block(config);
+    let db = db.clone().unsafe_into_genesis();
```

Comment:
Alright, so if I understand this correctly the reason we need to do this "unsafe" conversion is because we're patching a missing genesis state right? So this code path is only useful for existing nodes with the missing genesis tables, but once this is patched this code path is no longer needed?

Do we have a flow for defining database migrations? Having node operators run a separate migration against their persisted databases seems like an alternative approach here.

Reply:
yeah the insertion is idempotent though. tradeoff between one read-write and one write